commons/src/main/java/fr/free/nrw/commons/upload/APICalls.java

```diff
@@ -0,0 +1,169 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.android.volley.Cache;
+import com.android.volley.NetworkResponse;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.HttpHeaderParser;
+import com.android.volley.toolbox.JsonRequest;
+import com.android.volley.toolbox.Volley;
+
+import com.google.gson.Gson;
+
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class APICalls {
+
+    private static RequestQueue REQUEST_QUEUE;
+    private static final Gson GSON = new Gson();
+    private Context context;
+
+    public APICalls(Context context) {
+        this.context = context;
+    }
+    public void request(String apiUrl) {
+        JsonRequest<QueryResponse> request = new QueryRequest(apiUrl,
+                new LogResponseListener<QueryResponse>(), new LogResponseErrorListener());
+        getQueue().add(request);
+    }
+
+    private RequestQueue getQueue() {
+        return getQueue(context);
+    }
+
+    private static RequestQueue getQueue(Context context) {
```

Comment:
I think it would be safer if this function were synchronized to avoid a potential initialization race.